esx_service/vmdk_ops.py

```diff
@@ -702,6 +705,24 @@ def disk_attach(vmdk_path, vm):
     kv_status_attached, kv_uuid, attach_mode = getStatusAttached(vmdk_path)
     logging.info("Attaching {0} as {1}".format(vmdk_path, attach_mode))
 
+    # If the volume is attached then check if the attach is stale, if so detach
+    # the volume from the VM its attached to and attach to the requesting VM.
+    if kv_status_attached and kv_uuid != vm.config.uuid:
+       attached_to_vm = findVmByUuid(kv_uuid)
+       # Detach the disk only if VM is powered off and the disk
+       # is attached to it as the KV indicates.
+       device = findDeviceByPath(vmdk_path, attached_to_vm)
+       if attached_to_vm.runtime.powerState == VM_POWERED_OFF and device:
+          logging.warning("Detaching disk %s from VM uuid=%s\n", vmdk_path,
+                          attached_to_vm.config.uuid)
+          logging.warning("VM power state is %s\n",
+                          attached_to_vm.runtime.powerState)
```

Comment:
isn't it guaranteed to be down? Or are you concerned about race condition here so you want to double-check ? I think 1 log with "... from powered down VM uuid=" is good enough. Also, would be good to add VM names , since we have the vm.config object it should be easy